scripts/question_answering/model.py

```diff
@@ -0,0 +1,752 @@
+                self.convs.add(one_conv_module)
+
+        with self.name_scope():
+            self.dropout = gluon.nn.Dropout(opt.layers_dropout)
+            self.attention = SelfAttention(num_heads=num_heads)
+            self.attention_dropout = StochasticDropoutLayer(
+                (sub_layer_idx / total_layers) * (1 - opt.p_l))
+            sub_layer_idx += 1
+            self.attention_layer_norm = gluon.nn.LayerNorm(epsilon=1e-06)
+
+        self.positionwise_ffn = gluon.nn.HybridSequential()
+        with self.positionwise_ffn.name_scope():
+            self.positionwise_ffn.add(
+                gluon.nn.LayerNorm(epsilon=1e-06)
+            )
+            self.positionwise_ffn.add(
+                gluon.nn.Dropout(rate=opt.layers_dropout)
+            )
+            self.positionwise_ffn.add(
+                gluon.nn.Dense(
+                    units=opt.emb_encoder_conv_channels,
+                    activation='relu',
+                    use_bias=True,
+                    weight_initializer=MSRAPrelu(),
+                    flatten=False
+                )
+            )
+            self.positionwise_ffn.add(
+                gluon.nn.Dense(
+                    units=opt.emb_encoder_conv_channels,
+                    use_bias=True,
+                    weight_initializer=Xavier(),
+                    flatten=False
+                )
+            )
+            self.positionwise_ffn.add(
+                StochasticDropoutLayer(
+                    dropout=(sub_layer_idx / total_layers) * (1 - opt.p_l)
+                )
+            )
+
+    def hybrid_forward(self, F, x, mask):
+        r"""Implement forward computation.
+
+        Parameters
+        -----------
+        x : NDArray
+            input tensor with shape `(batch_size, sequence_length, hidden_size)`
+        mask : NDArray
+            input tensor with shape `(batch_size, sequence_length)`
+
+        Returns
+        --------
+        x : NDArray
+            output tensor with shape `(batch_size, sequence_length, hidden_size)`
+        mask : NDArray
+            output tensor with shape `(batch_size, sequence_length)`
+        """
+        x = self.position_encoder(x)
+        for conv in self.convs:
+            residual = x
+            x = conv(x) + residual
+        residual = x
+        x = self.attention_layer_norm(x)
+        x = F.Dropout(x, p=opt.layers_dropout)
+        x = self.attention(x, mask)
+        x = self.attention_dropout(x) + residual
+        return x + self.positionwise_ffn(x)
+
+
+class StochasticDropoutLayer(gluon.HybridBlock):
+    r"""
+    Stochastic dropout a layer.
+    """
+
+    def __init__(self, dropout, **kwargs):
+        super(StochasticDropoutLayer, self).__init__(**kwargs)
+        self.dropout = dropout
+        with self.name_scope():
+            self.dropout_fn = gluon.nn.Dropout(dropout)
+
+    def hybrid_forward(self, F, inputs):
+        if F.random.uniform().asscalar() < self.dropout:

```

Comment:
I don't think you can do this in a HybridBlock? Did you try hybridizing the model? Does it work?